packages/runtime/datastore-definitions/src/channel.ts

```diff
@@ -19,10 +20,17 @@ export interface IChannel extends IFluidLoadable {
     readonly attributes: IChannelAttributes;
 
     /**
+     * @deprecated - Use summarize instead.
      * Generates snapshot of the channel.
      */
     snapshot(): ITree;
 
+    /**
+     * Generates summary of the shared object.
+     * @returns A tree representing the summary of the channel and a set of nodes for garbage collection.
+     */
+    summarize(fullTree?: boolean): IChannelSummarizeResult;
```

Comment:
Making summarize async would provide some relief when trying to produce snapshots w/limited memory.

Reply:
You are right. summarize on IChannel has to eventually become async since in the future, it will be possible to have data stores in IChannel. However, I think we should do it a later PR where we will have to update summarize() in DDSs to return SummaryTree (or related PR). This PR is already too big and I would rather not add that here :).